internal/model/labels.go

```diff
@@ -59,6 +61,39 @@ func (l Labels) Clone() Labels {
 	return cp
 }
 
+// Equal returns true when the two maps of labels are equal.
+func (l Labels) Equal(labels Labels) bool {
+	if len(l) != len(labels) {
+		return false
+	}
+	for key, localV := range l {
+		v, ok := labels[key]
+		if !ok {
+			return false
+		}
+		// If the slice value is set, ignore the Value field.
+		if len(v.Values) == 0 && v.Value != localV.Value {
```

Comment:
Should we also check for `v.Global != localV.Global`? From the API, it seems possible for such a case to occur.

Reply:
I've moved the equality comparison to a private function inside the transaction aggregator where all the compared labels are global.